app/models/reviewable_chat_message.rb

```diff
@@ -4,6 +4,26 @@
 
 class ReviewableChatMessage < Reviewable
 
+  def self.on_score_updated(reviewable)
+    # Silence user if new score is over the `score_to_silence_user`
+    return if reviewable.type != self.name
+    return if reviewable.score <= self.score_to_silence_user
+
+    auto_silence_duration = SiteSetting.chat_auto_silence_from_flags_duration
```

Comment:
that's a nitpick but probably the first check we should do ?

Reply:
I don't know maybe I'm wrong, but I usually put SiteSetting check at top because it's mostly free and I don't have to think about what it could actually call (like score_to_silence_user could actually be a method doing some db call)